spec/integration/type/file_spec.rb

```diff
@@ -93,6 +93,43 @@ def get_aces_for_path_by_sid(path, sid)
     expect(Puppet::FileSystem.exist?(source)).to be_falsey
   end
 
+  describe "when ensure is present" do
+    before(:each) do
+      catalog.add_resource(described_class.new(:path => path, :ensure => :present, :backup => :false))
+    end
+
+    context "file is present" do
+      before(:each) do
+        FileUtils.touch(path)
+      end
+
+      it "should do nothing" do
+        report = catalog.apply.report
+        expect(report.resource_statuses["File[#{path}]"]).not_to be_failed
+        expect(Puppet::FileSystem.exist?(path)).to be_truthy
+      end
+
+      it "should log nothing" do
+        logs = catalog.apply.report.logs
+        expect(logs).to be_empty
+      end
+    end
+
+    context "file is not present" do
```

Comment:
`path` will be guaranteed to be recreated, because it uses `let(:path)` to create a `tmpfile`.